app/src/ui/lib/draggable.tsx

```diff
@@ -66,14 +68,19 @@ export class Draggable extends React.Component<IDraggableProps> {
    * - clears variables from last drag
    * - sets up mouse move and mouse up listeners
    */
-  private onMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
+  private onMouseDown = async (event: React.MouseEvent<HTMLDivElement>) => {
     if (!this.canDragCommit(event)) {
       return
     }
-    this.initializeDrag()
+    this.hasMouseUpOccurred = false
+    document.onmouseup = this.onMouseUp
+    await sleep(100)
+    if (this.hasMouseUpOccurred) {
+      return
+    }
 
+    this.initializeDrag()
     document.addEventListener('mousemove', this.onMouseMove)
-    document.onmouseup = this.onMouseUp
```

Comment:
I think what you're looking for here is a `debounce`. There should be one already accessible in the codebase thanks to `lodash`.

This implementation you have here basically ignores all quick clicks until the user keeps the mouse button down for more than 100ms. That's something super easy to model with `debounce`, I think, so you don't need to implement it yourself.

No need to address this right now or even for the upcoming beta/release, but I think it'd be nice if we explored that option and add that tool to our toolbox!

Basically I think it'd be something like this:
```typescript
private debouncedInitializeDrag = null
...
private onMouseDown = (event: React.MouseEvent<HTMLDivElement>) => { ```
    if (!this.canDragCommit(event)) {
      return
    }

	this.debouncedInitializeDrag = debounce(() => this.initializeDrag(), 100)
    document.addEventListener('mousemove', this.onMouseMove)
    document.onmouseup = this.onMouseUp
}
...
  private onMouseMove = (moveEvent: MouseEvent) => {
    if (this.debouncedInitializeDrag !== null) {
      this.debouncedInitializeDrag.cancel()
      this.debouncedInitializeDrag = null
      return
    }
    ...
  }
...
  private onMouseUp = (moveEvent: MouseEvent) => {
    if (this.debouncedInitializeDrag !== null) {
      this.debouncedInitializeDrag.cancel()
      this.debouncedInitializeDrag = null
      return
    }
    ...
  }
...
  private initializeDrag() {
    if (this.debouncedInitializeDrag !== null) {
      this.debouncedInitializeDrag.cancel()
      this.debouncedInitializeDrag = null
      return
    }
    ...
  }
```

Reply:
Interesting - I will look into it.

Reply:
I tried this.. and it doesn't seem to work.. debugger shows that it never invokes the method.
I originally had `_.debounce(()=>this.initializeDrag, 100)` as I think it should be for scoping only switched to picture to make sure that wasn't something weird. I double checked the wait time is in milliseconds. 

![image](https://user-images.githubusercontent.com/75402236/112679153-37cb7b00-8e42-11eb-956b-17f89d9e19d8.png)

Reply:
Also that this.debouncedDrag.cancel() crashes the app

Reply:
Sorry, I'm not familiar with how `debounce` works in JS/TS, only with the operation itself since I used it a lot in previous codebases, I just took a look at the API and it seemed the right way to use it 😅 

Again, that doesn't need to go in this PR, so don't worry!! 😄